Update fluent to 0.4.2

Hi Amethyst Crew!

I'm the maintainer of the `fluent-rs` crate and noticed that you're using an outdated version of our crate in your project `amethyst-locale`. We're advancing toward 1.0 and we updated the API in 0.4 which we hope we'll be able to keep toward stable release, and would appreciate if you could verify that it'll work for your project.

The update should be trivial I hope :)

Here's the compile error:

```
error[E0277]: `(dyn for<'r, 's> std::ops::Fn(&'r [std::option::Option<fluent::types::FluentValue>], &'s std::collections::HashMap<std::string::String, fluent::types::FluentValue>) -> std::option::Option<fluent::types::FluentValue> + 'static)` cannot be sent between threads safely
  --> amethyst_locale\src\lib.rs:47:6
   |
47 | impl Asset for Locale {
   |      ^^^^^ `(dyn for<'r, 's> std::ops::Fn(&'r [std::option::Option<fluent::types::FluentValue>], &'s std::collections::HashMap<std::string::String, fluent::types::FluentValue>) -> std::option::Option<fluent::types::FluentValue> + 'static)` cannot be sent between threads safely
   |
   = help: the trait `std::marker::Send` is not implemented for `(dyn for<'r, 's> std::ops::Fn(&'r [std::option::Option<fluent::types::FluentValue>], &'s std::collections::HashMap<std::string::String, fluent::types::FluentValue>) -> std::option::Option<fluent::types::FluentValue> + 'static)`
   = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<(dyn for<'r, 's> std::ops::Fn(&'r [std::option::Option<fluent::types::FluentValue>], &'s std::collections::HashMap<std::string::String, fluent::types::FluentValue>) -> std::option::Option<fluent::types::FluentValue> + 'static)>`
   = note: required because it appears within the type `std::boxed::Box<(dyn for<'r, 's> std::ops::Fn(&'r [std::option::Option<fluent::types::FluentValue>], &'s std::collections::HashMap<std::string::String, fluent::types::FluentValue>) -> std::option::Option<fluent::types::FluentValue> + 'static)>`
   = note: required because it appears within the type `fluent::entry::Entry<'static>`
   = note: required because it appears within the type `(std::string::String, fluent::entry::Entry<'static>)`
   = note: required because it appears within the type `std::marker::PhantomData<(std::string::String, fluent::entry::Entry<'static>)>`
   = note: required because it appears within the type `std::collections::hash::table::RawTable<std::string::String, fluent::entry::Entry<'static>>`
   = note: required because it appears within the type `std::collections::HashMap<std::string::String, fluent::entry::Entry<'static>>`
   = note: required because it appears within the type `fluent::bundle::FluentBundle<'static>`
   = note: required because it appears within the type `Locale`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `amethyst_locale`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Send/Sync is usually implemented automatically, but some API choices can prevent this from happening.  Here's some more info: https://doc.rust-lang.org/book/second-edition/ch16-04-extensible-concurrency-sync-and-send.html

If I find myself with some extra time I'll be happy to shoot a patch to you, but it also would probably just be faster to not bother waiting for me.

--- a/amethyst_locale/Cargo.toml
+++ b/amethyst_locale/Cargo.toml
@@ -22,7 +22,7 @@ travis-ci = { repository = "amethyst/amethyst" }
 [dependencies]
 amethyst_assets = { path = "../amethyst_assets", version = "0.4.0" }
 amethyst_core = { path = "../amethyst_core", version = "0.3.0" }
-fluent = "0.2.0"
+fluent = { git = "https://github.com/zbraniecki/fluent-rs", branch = "sendsync"}
 
 thread_profiler = { version = "0.3", optional = true }
 

--- a/amethyst_locale/Cargo.toml
+++ b/amethyst_locale/Cargo.toml
@@ -22,7 +22,7 @@ travis-ci = { repository = "amethyst/amethyst" }
 [dependencies]
 amethyst_assets = { path = "../amethyst_assets", version = "0.4.0" }
 amethyst_core = { path = "../amethyst_core", version = "0.3.0" }
-fluent = { git = "https://github.com/zbraniecki/fluent-rs", branch = "sendsync"}
+fluent = "0.4.3"
 
 thread_profiler = { version = "0.3", optional = true }
 

--- a/amethyst_locale/src/lib.rs
+++ b/amethyst_locale/src/lib.rs
@@ -9,7 +9,7 @@ extern crate fluent;
 
 use amethyst_assets::{Asset, Handle, ProcessingState, Result, SimpleFormat};
 use amethyst_core::specs::prelude::VecStorage;
-use fluent::MessageContext;
+use fluent::bundle::FluentBundle;
 
 /// Loads the strings from localisation files.
 #[derive(Clone)]
@@ -23,9 +23,9 @@ impl SimpleFormat<Locale> for LocaleFormat {
     fn import(&self, bytes: Vec<u8>, _: ()) -> Result<Locale> {
         let s = String::from_utf8(bytes)?;
 
-        let mut ctx = MessageContext::new(&[]);
-        ctx.add_messages(&s);
-        Ok(Locale { context: ctx })
+        let mut bundle = FluentBundle::new::<&'static str>(&[]);
+        bundle.add_messages(&s).expect("Error creating fluent bundle!");
+        Ok(Locale { bundle })
     }
 }
 
@@ -41,7 +41,7 @@ pub type LocaleHandle = Handle<Locale>;
 /// A loaded locale.
 pub struct Locale {
     /// The message context.
-    pub context: MessageContext<'static>,
+    pub bundle: FluentBundle<'static>,
 }
 
 impl Asset for Locale {

--- a/examples/locale/main.rs
+++ b/examples/locale/main.rs
@@ -65,18 +65,16 @@ impl<'a, 'b> SimpleState<'a, 'b> for Example {
                     println!(
                         "{}",
                         locale
-                            .context
-                            .get_message("hello")
-                            .and_then(|msg| locale.context.format(msg, None))
-                            .unwrap()
+                            .bundle
+                            .format("hello", None)
+                            .unwrap().0
                     );
                     println!(
                         "{}",
                         locale
-                            .context
-                            .get_message("bye")
-                            .and_then(|msg| locale.context.format(msg, None))
-                            .unwrap()
+                            .bundle
+                            .format("bye", None)
+                            .unwrap().0
                     );
                 }
             }

